test/whitelisted_crowdsale.js

```diff
@@ -73,3 +76,77 @@ contract('WhitelistedCrowdsale', async (accounts) => {
   });
 
 });
+
+contract('WhitelistedCrowdsale', async (accounts) => {
+  let owner = accounts[0];
+  let holder = accounts[1];
+  let escrow_account = accounts[2];
+  let user = accounts[3];
+  let userBalance = web3.eth.getBalance(user);
+  let defaultRateForPrivateOffer = 14000;
+
+  let controller = null;
+  let instanceWhitelistedCrowdsale = null;
+  let token = null;
+
+  beforeEach("setup contract for each test", async () => {
+    let startTime = Math.ceil(Date.now() / 1000);
+    let endTime = Math.ceil(Date.now() / 1000) + 100;
+    
+    controller = await ICO_controller.new(holder, escrow_account, {from: owner});
+    token = MFC_Token.at(await controller.token.call());
+    await controller.startPrivateOffer.sendTransaction(startTime, endTime, escrow_account, {from: owner});
+    instanceWhitelistedCrowdsale = WhitelistedCrowdsale.at(await controller.privateOffer.call());
+    wait(5);
+  });
+
+  afterEach("clean contracts" , async () => {
+    controller = null;
+    instanceWhitelistedCrowdsale = null;
+    token = null;
+  });
+  
+  it("should be deployed", async () => {
+    assert.isOk(instanceWhitelistedCrowdsale);
+  });
+
+  it("should throw error when beneficiary not in whitelist", async () => {
+    try {
+      await instanceWhitelistedCrowdsale.buyTokens.sendTransaction(user, {from: owner, value: web3.toWei(2, "ether")});
+    } catch (err) {
+      assert.equal(err, 'Error: VM Exception while processing transaction: revert', "Benefeciary not in whitelist trying to buy tokens");
+    }
+  });
+
+  it("should throw error when beneficiary has a zero address", async () => {
```

Comment:
I think it would be clearer to say `should throw an error when beneficiary is the ICO owner`

Reply:
In this case we check line 
require(beneficiary != address(0));

Reply:
@avgoeid that is right, but isn't the `address(0)` is the address of the owner, the account that has created the contract?